### PR TITLE
Feat/mzad 26/header

### PR DESCRIPTION
**Pull request by Keretcha**

### CODE REVIEW

File: app/Components/Header/HeaderNavigation/HeaderNavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import NavMenu from '../../NavMenu/NavMenu';
+import styles from './HeaderNavigation.module.scss';
+
+const HeaderNav = (): JSX.Element => {
+  return (
+    <div className={styles.container}>
+      <div className={styles.navigation}>
+        <NavMenu
+          items={[
+            {
```
Comment (pailodze) [inline app/Components/Header/HeaderNavigation/HeaderNavigation.tsx:11]: Why do we need HeaderNavigation.tsx as Another Component?

File: app/States/States.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { RecoilState, atom } from 'recoil';
+
+export const isDarkState: RecoilState<boolean> = atom({
+  key: 'dark',
```
Comment (pailodze) [inline app/States/States.ts:4]: key should be isDark

File: app/States/States.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  default: false,
```
Comment (pailodze) [inline app/States/States.ts:5]: default should come from localStorage

File: app/States/States.ts (hunk 2 above)
Comment (pailodze) [inline app/States/States.ts:4]: States.ts Should be states.ts

File: app/Components/Header/Interfaces/Header-props.interface.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface HeaderPropsInterface {
+  mode?: boolean;
```
Comment (pailodze) [inline app/Components/Header/Interfaces/Header-props.interface.ts:2]: this props should me isDark

**Review by pailodze — CHANGES_REQUESTED** [review]

File: app/Components/Header/HeaderNavigation/HeaderNavigation.tsx (hunk 1 above)
Comment (Keretcha) [reply app/Components/Header/HeaderNavigation/HeaderNavigation.tsx:11]: cause in header component its not readable

File: app/Components/Header/Mode/Mode.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import { useRecoilState } from 'recoil';
+import Icon from '../../Icon/Icon';
+import { IconNameEnum } from '../../Icon/enums/icon-name.enum';
+import styles from './Mode.module.scss';
+import { isDarkState } from '@/app/States/states';
+
+const Mode = (): JSX.Element => {
+  const [dark, setDark] = useRecoilState(isDarkState);
```
Comment (pailodze) [inline app/Components/Header/Mode/Mode.tsx:10]: State Usage Shoudl Be At Top

File: app/Components/Header/HeaderNavigation/HeaderNavigation.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// import NavMenu from '../../NavMenu/NavMenu';
```
Comment (pailodze) [inline app/Components/Header/HeaderNavigation/HeaderNavigation.tsx:1]: Remove this component file its uesless code

File: app/Components/Header/Interfaces/Header-props.interface.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export interface HeaderPropsInterface {
+  isDark?: boolean;
```
Comment (pailodze) [inline app/Components/Header/Interfaces/Header-props.interface.ts:2]: name should be header-props.interface.ts

File: app/Components/Header/ModeSwitcher/ModeSwitcher.module.scss [hunk 8]
```diff
@@ -0,0 +1,15 @@
+@import '../../../Colors/common-colors.scss';
+@import '../../../Colors/dark-colors.scss';
+@import '../../../Colors/light-colors.scss';
+
```
Comment (pailodze) [inline app/Components/Header/ModeSwitcher/ModeSwitcher.module.scss:4]: remove this spaces

File: app/Components/NavMenu/NavMenu.module.scss [hunk 9]
```diff
@@ -4,26 +4,22 @@
 .container {
   display: flex;
   flex-flow: row wrap;
-  flex-basis: 100%;
+  // flex-basis: 100%;
```
Comment (pailodze) [inline app/Components/NavMenu/NavMenu.module.scss:7]: remove commmeent

File: app/States/States.ts (hunk 2 above)
Comment (pailodze) [reply app/States/States.ts:4]: man key should be isDark!!!

Comment (pailodze) [reply app/States/States.ts:4]: States ts should be states.ts

**Review by pailodze — CHANGES_REQUESTED** [review]

File: app/Components/Header/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+const Header: HeaderType = () => {
+  const [dark, setDark] = useRecoilState(isDarkState);
+
+  useEffect(() => {
+    localStorage.setItem('isDark', String(dark));
+
+    const isDarkMode: boolean = localStorage.getItem('isDark') === 'true';
+
+    if (isDarkMode !== dark) {
+      setDark(isDarkMode);
```
Comment (pailodze) [inline app/Components/Header/Header.tsx:24]: what is this logic?

File: app/Components/Header/Interfaces/Header-props.interface.ts (hunk 7 above)
Comment (pailodze) [reply app/Components/Header/Interfaces/Header-props.interface.ts:2]: Name is still same

File: app/Components/Header/Type/Header.type.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+import { HeaderPropsInterface } from '../Interfaces/header-props.interface';
```
Comment (pailodze) [inline app/Components/Header/Type/Header.type.ts:1]: File Name Should Be header.type.ts

File: app/States/States.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import { RecoilState, atom } from 'recoil';
+
+export const isDarkState: RecoilState<boolean> = atom({
+  key: 'isDark',
+  default:
```
Comment (pailodze) [inline app/States/States.ts:5]: dont think we need it anymore

**Review by pailodze — CHANGES_REQUESTED** [review]

File: app/Components/Header/Header.tsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+import { HeaderType } from './Type/header.type';
+
+const Header: HeaderType = () => {
+  const [dark, setDark] = useRecoilState(isDarkState);
+
+  useEffect(() => {
+    setDark(localStorage.getItem('isDark') === 'true');
+  }, [setDark]);
+
+  console.log(dark);
```
Comment (pailodze) [inline app/Components/Header/Header.tsx:22]: Remove Console Logs

**Review by pailodze — CHANGES_REQUESTED** [review]

**Pull request merged**